main_test.go

```diff
@@ -157,6 +157,60 @@ func TestGenerate(t *testing.T) {
     }
   },
   "server_listener_resource_name_template": "grpc/server?xds.resource.listening_address=%s"
+}`,
+		},
+		{
+			desc: "happy case with deployment info",
+			input: configInput{
+				xdsServerUri:      "example.com:443",
+				gcpProjectNumber:  123456789012345,
+				vpcNetworkName:    "thedefault",
+				ip:                "10.9.8.7",
+				zone:              "uscentral-5",
+				includeV3Features: true,
+				deploymentInfo: map[string]string{
+					"GCP-ZONE":      "uscentral-5",
+					"GKE-CLUSTER":   "test-gke-cluster",
+					"GKE-NAMESPACE": "test-gke-namespace",
+					"GKE-POD":       "test-gke-pod",
+					"INSTANCE-IP":   "10.9.8.7",
+					"GKE-VM":        "test-gce-vm",
```

Comment:
QQ: Is GKE-VM this a typo? In main.go, i only see GCP-VM (i left a comment about changing it to GCE-VM). 
How did this end up in node metadata, event though "GKE-VM" is unknown to main.go?

Reply:
So, this test only tests the `generate` function. It doesn't test thing end to end. But I've fixed it here and in main.go to use `GCE-VM` instead. Thanks.